src/interpolation/details/ArborX_InterpDetailsPolynomialBasis.hpp

```diff
@@ -0,0 +1,173 @@
+// and dimension. For example, the polynomial basis of degree 2 evaluated at a
+// point {x, y, z} would be [1, x, y, z, xx, xy, yy, xz, yz, zz].
+//
+// To compute it, the list of values is sliced against its degree and highest
+// dimension. From the previous list, we would have the following sublists:
+// - [1]                             at degree 0
+// - [x], [y] and [z]                at degree 1
+// - [xx], [xy, yy] and [xz, yz, zz] at degree 2
+// One can then infer a recursive pattern that will be used to build the list:
+// - [1]                             at degree 0
+// - x*[1], y*[1] and z*[1]          at degree 1
+// - x*[x], y*[x, y] and z*[x, y, z] at degree 2
+// So, given a slice at degree n and dimension u, its values would be the
+// product of all the slices of degree n-1 and of dimension u or less with the
+// coordinate of dimension u.
+//
+// As another example, if we can take the polynomial basis of degree 3 evaluated
+// at a point {x, y} would be [1, x, y, xx, xy, yy, xxx, xxy, xyy, yyy]. Its
+// slices are:
+// - [1]                       at degree 0
+// - [x] and [y]               at degree 1
+// - [xx] and [xy, yy]         at degree 2
+// - [xxx] and [xxy, xyy, yyy] at degree 3
+// And its recursive pattern:
+// - [1]                       at degree 0
+// - x*[1] and y*[1]           at degree 1
+// - x*[x] and y*[x, y]        at degree 2
+// - x*[xx] and y*[xx, xy, yy] at degree 3
+//
+// These examples can be represented in 2D as the following tables:
+//   Quadratic |  3D  ||        Cubic      |  2D
+// ------------+----- || ------------------+-----
+//    degree   | dim  ||       degree      | dim
+// ---+---+----+      || ---+---+----+-----+
+//  0 | 1 |  2 |      ||  0 | 1 |  2 |  3  |
+// ===o===o====o      || ===o===o====o=====o
+//  1 |        |      ||  1 |              |
+// ---+---+----+----- || ---+---+----+-----+-----
+//    | x |    |  x   ||    | x |    |     |  x
+//    |   | xx |      ||    |   | xx |     |
+//    +---+----+----- ||    |   |    | xxx |
+//    | y |    |  y   ||    +---+----+-----+-----
+//    |   | xy |      ||    | y |    |     |  y
+//    |   | yy |      ||    |   | xy |     |
+//    +---+----+----- ||    |   | yy |     |
+//    | z |    |  z   ||    |   |    | xxy |
+//    |   | xz |      ||    |   |    | xyy |
+//    |   | yz |      ||    |   |    | yyy |
+//    |   | zz |      ||
+
+// This function returns the lengths of the slices for a given maximum degree
+// and dimension.
+// Given a slice at degree n and dimension u, its length would be the sum of all
+// the slices' lengths of degree n-1 and of same or smaller dimension.
+//
+// Given the previous two examples, the two arrays would be:
+// Deg \ Dim | x | y | z  || Deg \ Dim | x | y
+// ----------+---+---+--- || ----------+---+---
+//      1    | 1 | 1 | 1  ||      1    | 1 | 1
+//      2    | 1 | 2 | 3  ||      2    | 1 | 2
+//                        ||      3    | 1 | 3
+template <std::size_t Dim, std::size_t Deg>
```

Comment:
```suggestion
template <std::size_t DIM, std::size_t Degree>
```
We use `DIM` in other places, so it would be nice for consistency.
And I think there is little reason to shorten `Degree`, and it would be easier to read.